libc/newhdrgen/yaml_combined/ctype.yaml

```diff
@@ -0,0 +1,135 @@
+header: ctype.h
+macros: []
+types: []
+enums: []
+objects: []
+functions:
+  - name: isalnum
+    standards: 
+      - stdc
+    return_type: int
+    arguments:
+      - type: int
+    guard: null
```

Comment:
in future it'd be good to make it so that if an optional member (e.g. `guard`) isn't listed it defaults to `null` or an empty list or similar to cut down on boilerplate. It's not necessary for this patch but it's good to consider for the future.